backend/src/main/java/de/codexbella/ContentService.java

```diff
@@ -86,28 +87,98 @@ public Optional<Show> getShow(String showId, String username) {
                      +"&language="+language, String.class);
          SeasonApi seasonApi = new Gson().fromJson(response, SeasonApi.class);
          Season season = contentMapper.toSeason(seasonApi, username);
-         season.setNumberOfEpisodes(showOptional.get().getSeasons().get(seasonNumber-1).getNumberOfEpisodes());
+         season.setNumberOfEpisodes(seasonBefore.getNumberOfEpisodes());
+         season.setSeen(seasonBefore.getSeen());
+         if (seasonBefore.getSeen() == Seen.YES) {
+            for (Episode episode : season.getEpisodes()) {
+               episode.setSeen(Seen.YES);
+            }
+         }
+         season.setRating(seasonBefore.getRating());
          showOptional.get().getSeasons().set(seasonNumber-1, season);
          showRepository.save(showOptional.get());
       }
       return showOptional;
    }
 
-   public Optional<Show> editShow(String showId, int rating, Integer seasonNumber, Integer episodeNumber, String username) {
+   public Optional<Show> editShow(String showId, Integer rating, Seen seen, Integer seasonNumber, Integer episodeNumber, String username) {
       Optional<Show> showOptional = showRepository.findByIdAndUsername(showId, username);
       if (showOptional.isPresent()) {
          Show show = showOptional.get();
-         if (episodeNumber != null) {
-            show.getSeasons().get(seasonNumber-1).getEpisodes().get(episodeNumber-1).setRating(rating);
-         } else if (seasonNumber != null) {
-            show.getSeasons().get(seasonNumber-1).setRating(rating);
-         } else {
-            show.setRating(rating);
+         if (rating != null) {
+            if (episodeNumber != null) {
+               show.getSeasons().get(seasonNumber-1).getEpisodes().get(episodeNumber-1).setRating(rating);
+            } else if (seasonNumber != null) {
+               show.getSeasons().get(seasonNumber-1).setRating(rating);
+            } else {
+               show.setRating(rating);
+            }
+         }
+
+         if (seen != null) {
+            if (episodeNumber != null) {
+               Season season = show.getSeasons().get(seasonNumber-1);
+               show.getSeasons().get(seasonNumber-1).getEpisodes().get(episodeNumber-1).setSeen(seen);
+               if (seen == Seen.NO) {
+                  if (season.getEpisodes().stream().allMatch(e -> e.getSeen() == Seen.NO)) {
+                     season.setSeen(Seen.NO);
+                  } else {
+                     season.setSeen(Seen.PARTIAL);
+                  }
+                  if (show.getSeasons().stream().allMatch(s -> s.getSeen() == Seen.NO)) {
+                     show.setSeen(Seen.NO);
+                  } else {
+                     show.setSeen(Seen.PARTIAL);
+                  }
+               } else if (seen == Seen.YES) {
+                  if (season.getEpisodes().stream().allMatch(e -> e.getSeen() == Seen.YES)) {
+                     season.setSeen(Seen.YES);
```

Comment:
puh das ist ziemlich viel...wenn ich das richtig verstehe, überprüfst du hier ob eine oder mehrere Staffeln schon geschaut wurden und setzt dann dies auf NO, YES oder PARTIAL. Das würde sich auf jeden Fall schon mal anbieten, solche Parts in eigene Funktionen auszulagern.